FEAT :: packages ui - Inputs, Button, TextAreas

j02on opened the pull request:
packages ui에 Inputs, Button, TextAreas를 추가 하였습니다.

j02on committed:
FEAT :: packages ui - Inputs, Button, TextAreas

packages ui에 Inputs, Button, TextAreas를 추가 하였습니다.

dutexion commented on package.json:
```diff
@@ -32,6 +32,7 @@
     "@emotion/react": "11.14.0",
     "@emotion/styled": "11.14.0",
     "@entry/design-token": "*",
+    "@entry/ui": "*",
     "@types/react-query": "1.2.9",
     "react": "18.3.1",
     "react-dom": "18.3.1",
```
공통으로 사용되지 않는 의존성들을 지워야 할 것 같아요

dutexion commented on packages/ui/Button.tsx:
```diff
@@ -0,0 +1,32 @@
+import { plus } from "./assets";
+
+interface button {
+  isAdmin?: "admin" | "user";
```
is <- 이거는 boolean 타입에 주로 사용합니다..

dutexion commented on packages/ui/Button.tsx:
```diff
@@ -0,0 +1,32 @@
+interface button {
+  isAdmin?: "admin" | "user";
+  children?: string;
+  type?: "addChildren" | "textChildren";
+}
+
+export const Button = ({ isAdmin, children, type }: button) => {
+  return (
+    <ButtonContainer type="button" isAdmin={isAdmin}>
+      {type == "addChildren" ? <img src={plus} alt="plusImg" /> : children}
+    </ButtonContainer>
+  );
+};
+
```
더욱더 구조화 된 코드를 작성하기 위해서는
아이콘 또한 prop으로 받아서 처리하는 방식도 있습니다

dutexion commented on packages/ui/Button.tsx:
```diff
@@ -0,0 +1,32 @@
+
+export const Button = ({ isAdmin, children, type }: button) => {
+  return (
+    <ButtonContainer type="button" isAdmin={isAdmin}>
```
isAdmin 파라미터는 boolean으로 정의 되어 있는데,
Button의 isAdmin 파라미터는 유니온 타입입니다.

dutexion commented on packages/ui/Button.tsx:
```diff
@@ -0,0 +1,32 @@
+import styled from "@emotion/styled";
+import { plus } from "./assets";
+
+interface button {
```
type 명을 소문자로 시작하게 되면 일반 변수와 헷갈리는 경우가 생길 수도 있습니다.
민수님과 상의하여 타입명 컨벤션을 지정하는 것도 좋을 것 같아요

또한 아무 상속도 받지 않고 상속을 하지 않는 것에 대해서는
type alias가 나은 선택일 수 있습니다.

dutexion commented on packages/ui/Button.tsx:
```diff
@@ -0,0 +1,32 @@
+  );
+};
+
+const ButtonContainer = styled.button<{ isAdmin: boolean }>`
```
isAdmin이라는 타입에 대해서 `interface button`에 있는 isAdmin과 동일한 타입을 사용하려면
pick 혹은 omit 문법을 통한 멋진 코드를 작성할 수 있습니다.

dutexion commented on packages/ui/assets/plus.svg:
svg 아이콘을 곧이곧대로 사용하지 않고,
tsx 컴포넌트화를 통하여 사이즈와 색상을 조정할 수 있도록 하여
재사용성을 높일 수 있습니다..

dutexion commented on packages/ui/TextAreas.tsx:
```diff
@@ -0,0 +1,59 @@
+export const TextAreas = ({ label, type, placeholder }: textareas) => {
+  return (
+    <TextAreaContainer>
+      {type == "labelInput" ? (
+        <LabelInput placeholder={placeholder} />
+      ) : (
+        <LabelText>{label}</LabelText>
+      )}
```
Input과 textArea의 label 부분이 동일한 기능으로 보이는데,
이 부분 또한 컴포넌트화 하는게 더 좋을 것 같아요

dutexion commented on packages/ui/Inputs.tsx:
```diff
@@ -0,0 +1,58 @@
+export const Inputs = ({ label, type, placeholder }: inputs) => {
+  return (
+    <InputContainer>
+      {type == "labelInput" ? (
+        <LabelInput placeholder={placeholder} />
+      ) : (
+        <LabelText>{label}</LabelText>
+      )}
```
중첩 되지 않은 삼항 연산자에 사용하는 type
즉, 유니온이 두 개뿐만이라면 boolean 타입으로 작성하는 것도 괜찮을 듯 싶습니다.

제가 이해한 바로는 label 내용을 변경할 수 있냐 없냐에 따라서 변경되는 내용 같은데,
isFixedLabel 혹은 isCanChange 등등을 사용할 수 있을 것 같습니다

j02on committed:
FIX :: 리뷰 반영 수정

icon tsx 컴포넌트화
Button 타입 pick 사용하여 수정
상속 받지 않는 interface -> type alias 로 수정
type명 대문자로 수정
Label 컴포넌트화
root package.json 공통으로 사용되지 않는 의존성 삭제

dutexion commented:
리뷰어에 민수님도 넣어주세요.

10기 끼리도 코드 리뷰하면 좋을거 같아요
